ui: Replace with core-ui tablev2 component

ChengYanJin opened the pull request:
**Component**: UI

**Context**: 
Bump core-ui to 0.41.0
Cypress version to 6.1.0

Replace the following tables with tablev2
- [x] Node List Table
- [x] Node Alert Table
- [x] Node Partition Table
- [x] Volume List Table
- [x] Volume Alert Table


**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui: Replace the NodePartition Table with Tablev2

ChengYanJin committed:
ui: Replace the Volume Alert Table with tablev2

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/metalk8s/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @lucieleonard

* @jbwatenbergscality

* @chengyanjin

* @thomasdanan

* @gdemonet

* @teddyandrieux

ChengYanJin committed:
ui: Bump the core-ui version with 0.41.0

ChengYanJin committed:
ui: Replace NodeList Table with tablev2

ChengYanJin committed:
ui: Replace the Volume List Table with tablev2

ChengYanJin committed:
ui: Replace the Node Alert Table with tablev2

ChengYanJin committed:
ui: Remove the filter when we check volume from node page

JBWatenbergScality commented on ui/src/components/VolumeAlertsTab.js:
```diff
@@ -1,100 +1,13 @@
-
-const ActiveAlertsCard = (props) => {
-  const { alertlist, PVCName } = props;
+const ActiveAlertsCard = ({ alertlist, PVCName }) => {
```
nit: maybe we can add typing by the way

JBWatenbergScality commented on ui/src/components/VolumeAlertsTab.js:
```diff
@@ -115,133 +28,68 @@ const ActiveAlertsCard = (props) => {
+                    {intl.formatMessage({ id: 'volume_is_not_bound' })}
+                  </NoResult>
+                );
+              } else if (PVCName && alertlist.length === 0)
+                return (
+                  <NoResult>
+                    {intl.formatMessage({ id: 'no_active_alerts' })}
+                  </NoResult>
+                );
```
nit nit nit
```suggestion
} else if (PVCName && alertlist.length === 0) {
                return (
                  <NoResult>
                    {intl.formatMessage({ id: 'no_active_alerts' })}
                  </NoResult>
                );
                }
```

JBWatenbergScality commented on ui/src/components/NodeListTable.js:
```diff
@@ -420,25 +36,38 @@ const NodeListTable = (props) => {
+            <>
+              <LargeText data-cy="node_table_name_cell">{name}</LargeText>
+              <Box display={'inline-flex'} flexWrap={'wrap'}>
+                {controlPlaneIP ? <IPText>CP : {controlPlaneIP}</IPText> : null}
```
We should try to use an "official" text component or maybe split the info in 2 additional columns.

JBWatenbergScality commented on ui/src/components/VolumeListTable.js:
```diff
@@ -490,32 +153,19 @@ const VolumeListTable = (props) => {
-    if (isAddNodeFilter || !isNodeColumn) {
-      history.push(`/volumes/${row.values.name}/overview?node=${nodeName}`);
+    if (isAddNodeFilter) {
+      history.push(`/volumes/${row.values.name}/overview?search=${nodeName}`);
```
```suggestion
      history.push(`/volumes/${row.values.name}/overview?${query.toString()}`);
```
and above condition can be dropped

JBWatenbergScality commented on ui/src/components/VolumeListTable.js:
```diff
@@ -536,17 +186,52 @@ const VolumeListTable = (props) => {
+            icon={<i className="fas fa-plus" />}
+            onClick={() => {
+              // depends on if we add node filter
+              if (nodeName) {
```
To be checked if the property is still used but it seems not so maybe this can be deleted

JBWatenbergScality commented on ui/src/components/VolumeListTable.js:
```diff
@@ -536,17 +186,52 @@ const VolumeListTable = (props) => {
+          selectedId={volumeName}
+          onRowSelected={onClickRow}
+          children={(Rows) => {
+            return <>{Rows}</>;
```
I think an empty state is missing here. It should likely take the same content of loki/prom/alert manager volume missing.

bert-e commented:
# Conflict




There is a conflict between your branch `feature/deprecated-orphan-coreui-components` and the
destination branch `development/124.0`.

Please resolve the conflict on **the feature branch** (`feature/deprecated-orphan-coreui-components`).

```sh
git fetch && \
git checkout origin/feature/deprecated-orphan-coreui-components && \
git merge origin/development/124.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/deprecated-orphan-coreui-components
```

ChengYanJin committed:
shell-ui: bump core-ui version to 0.41.0

ChengYanJin committed:
ui: Extract AlertsTab

ChengYanJin committed:
ui: remove the node filter when selecting a volume

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @lucieleonard

* @jbwatenbergscality

* @chengyanjin

* @thomasdanan

* @gdemonet

* @teddyandrieux

ChengYanJin committed:
ui: Replace the CP and IP in Node list with SecondaryText

ChengYanJin committed:
ui: Remove the unnecessary re-renders

ChengYanJin committed:
ui: Fix the integration tests

ChengYanJin committed:
ui: Add mockOffsetSize to render the virtualized component

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @gdemonet

* @lucieleonard

* @teddyandrieux

* @jbwatenbergscality

* @chengyanjin

ChengYanJin committed:
ui: Bump the Cypress version to v6.1.0

ChengYanJin committed:
ui/test: Replace cy.route2() with cy.intercept()

Fix the breaking change:
JSON request and response bodies captured by cy.intercept() are now automatically parsed, removing the need to manually do JSON.parse. Addressed in #9280.

ChengYanJin committed:
ui/test: Fix the volumelist table test

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/124.0`



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/124.0`



The following branches have **NOT** changed:

* `development/123.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye chengyanjin.
